numpy/lib/utils.py

```diff
@@ -193,7 +193,41 @@ def deprecate(*args, **kwargs):
     else:
         return _Deprecate(*args, **kwargs)
 
-deprecate_with_doc = lambda msg: _Deprecate(message=msg)
+
+def deprecate_with_doc(msg):
+    """
+    Decorator to deprecate a function and change its docstring to
+    mention the deprecation. 
+    
+    It returns an object that can be used to Issue a DeprecationWarning, 
+    by passing `func` as argument, this adds warning to the to-be 
+    decorated function's docstring and returns the new function object.
+    
+    See Also
+    --------
+    deprecate : Decorate a function such that it issues a `DeprecationWarning` 
+    
+    Parameters
+    ----------
+    msg : str
+    	    Additional explanation of the deprecation. Displayed in the 
+       	    docstring after the warning.
+
+    Returns
+    -------
+    obj : object
+
+    Examples
+    --------
+    Note that ``olduint`` returns a value after printing DeprecationWarning 
```

Comment:
The example doesn't actually show the deprecation warning, can you fix that?

Reply:
Also, `printing` -> `raising`

Reply:
Fixed.

Reply:
deprecation warning is still not shown

Reply:
The warning has been added however I believe that this is causing 3 of the azure-pipeline tests to fail.

Reply:
Those tests will fail until gh-17897 has been merged, there was a small merge conflict with two PRs.

Reply:
Also, should the deprecation warning for `numpy.deprecate` ([docs](https://numpy.org/doc/stable/reference/generated/numpy.deprecate.html?highlight=numpy%20deprecate#numpy.deprecate)) also be moved to after the call to `olduint` in the example?

Reply:
> Those tests will fail until [gh-17897](https://github.com/numpy/numpy/pull/17897) has been merged, there was a small merge conflict with two PRs.

Alright